src/memory/buffer_pool.cpp

```diff
@@ -4,8 +4,8 @@
 
 SMILE_NS_BEGIN
 
-BufferPool::BufferPool(FileStorage* storage, const BufferPoolConfig& config) noexcept {
-	p_storage = storage;
+BufferPool::BufferPool( const FileStorage& storage, const BufferPoolConfig& config ) noexcept {
```

Comment:
FileStorage should be a pointer (FileStorage*) as was before, since this is effectively an object that whose state is modified from within the buffer pool (e.g. the file can grow up when reserving more space). Also, removing the const qualifier from a const object is a very dangerous operation that could lead to an undefined behavior, be careful!

Reply:
In fact I was not really confident with that change. Thanks!